### DISCUSSION TITLE
How to process batch images in CNN using FINN?

### DISCUSSION THREAD

**Discussion by Arbiter-glitch** [Xilinx/finn#1029] [Q&A] [answered]
Iam a beginner in finn. Using finn v0.9. I tried to export for a batch size of 8 but my network uses an upsampling layer so while applying the MakeScaleResizeNHWC() transformation the tranpose nodes that get added do not accurately apply it. That is in my case if the input before the tranpose node is 8,1,40,40 instead of it becoming 8,40,40,1 it becomes 1,40,40,1. Is there a way to rectify it?
The mode after tidying up:-
![model onnx](https://github.com/Xilinx/finn/assets/70324978/0d233bd2-2570-42d8-8d9a-9046e016545a)
The model after streamline and makescaleresize:-
![fsrcnn-check onnx](https://github.com/Xilinx/finn/assets/70324978/cbeba0a9-45e0-4956-af95-999a0c5c7581)

**Comment by fpjentzsch** [comment, marked as answer]
Hi,
batch size is not really a thing for the dataflow accelerators FINN generates. They operate on a continuous stream of input data, one frame after the other, so the batch size is essentially infinite. For this reason, N=1 is assumed in many places and you should export your model with batch size = 1.

The PYNQ driver FINN generates for you still supports batching, i.e., you can give a batch of frames to the driver and it will stream them frame by frame to the accelerator for processing.

**Reply by Arbiter-glitch** [reply]
Thank you sir... I have one more query. In the newest finn v0.10, there is support for deconvolution. But the output padding feature of deconvolution is not infered. Is that feature of the pytorch convtranspose2d layer not incorporated yet?
